Xplat/src/main/java/vazkii/botania/common/block/block_entity/RunicAltarBlockEntity.java

```diff
@@ -232,11 +233,12 @@ private void saveLastRecipe() {
 		level.blockEvent(getBlockPos(), BotaniaBlocks.runeAltar, SET_KEEP_TICKS_EVENT, 400);
 	}
 
-	public void trySetLastRecipe(Player player) {
+	public InteractionResult trySetLastRecipe(Player player) {
 		InventoryHelper.tryToSetLastRecipe(player, getItemHandler(), lastRecipe, null);
 		if (!isEmpty()) {
 			VanillaPacketDispatcher.dispatchTEToNearbyPlayers(this);
 		}
+		return lastRecipe == null || lastRecipe.isEmpty() ? InteractionResult.CONSUME : InteractionResult.SUCCESS;
```

Comment:
I think you want PASS instead of CONSUME here. Currently, if your main hand is empty and your offhand has a block, right clicking a watered apothecary or an altar will do nothing, which feels really awkward. I'd expect it to place the block (like what happens with an empty apothecary).

Reply:
correct